scene/3d/spring_arm.cpp

```diff
@@ -93,7 +93,10 @@ void SpringArm::set_length(float p_length) {
 	if (is_inside_tree() && (Engine::get_singleton()->is_editor_hint() || get_tree()->is_debugging_collisions_hint()))
 		update_gizmo();
 
-	spring_length = p_length;
+	if (p_length != spring_length) {
```

Comment:
You could move this at the start of the method as
```cpp
if (p_length != sprint_length) {
	return;
}
```
so that we don't update the gizmo needlessly either.